test/sql/window/test_window_dbplyr.test

```diff
@@ -15,15 +15,15 @@ INSERT INTO dbplyr_052 VALUES (1,1, 42),(2,1, 42),(3,1, 42),(2,2, 42),(3,2, 42),
 
 # this works fine because we order by the already-projected column in the innermost query
 query IR
-SELECT x, g FROM (SELECT x, g, SUM(x) OVER (PARTITION BY g ORDER BY x ROWS UNBOUNDED PRECEDING) AS zzz67 FROM (SELECT x, g FROM dbplyr_052 ORDER BY x) dbplyr_053) dbplyr_054 WHERE (zzz67 > 3.0)
+SELECT x, g FROM (SELECT x, g, SUM(x) OVER (PARTITION BY g ORDER BY x ROWS UNBOUNDED PRECEDING) AS zzz67 FROM (SELECT x, g FROM dbplyr_052 ORDER BY x) dbplyr_053) dbplyr_054 WHERE (zzz67 > 3.0) ORDER BY x, g
 ----
 3	1.000000
 3	2.000000
 4	2.000000
 
 # this breaks because we add a fake projection that is not pruned
 query IR
-SELECT x, g FROM (SELECT x, g, SUM(x) OVER (PARTITION BY g ORDER BY x ROWS UNBOUNDED PRECEDING) AS zzz67 FROM (SELECT x, g FROM dbplyr_052 ORDER BY w) dbplyr_053) dbplyr_054 WHERE (zzz67 > 3.0)
+SELECT x, g FROM (SELECT x, g, SUM(x) OVER (PARTITION BY g ORDER BY x ROWS UNBOUNDED PRECEDING) AS zzz67 FROM (SELECT x, g FROM dbplyr_052 ORDER BY w) dbplyr_053) dbplyr_054 WHERE (zzz67 > 3.0) ORDER BY x, g
```

Comment:
Are there any tests for the parallel window functions? It would be great if you could add some under `test/sql/parallelism/intraquery`. The SQL server tests (`test\sqlserver\test_sqlserver.test_slow`) would be a good starting point as they contain many window functions. You can add the following lines on top there to force the parallel window functions to be tested:

```sql
statement ok
PRAGMA threads=4

statement ok
PRAGMA force_parallelism
```

It would be great to see some more simple/stand-alone tests as well.

Reply:
I can do that, but to some extent I was assuming that my changes were a NOP. The test that @hannesmuehleisen found above seems to be a good example (although I think it may be too big?)

Reply:
On the contrary I would think introducing parallelism to an operator is an easy way to introduce many subtle bugs :) I would prefer to see this tested well.

Reply:
`test/sql/window/test_over_grouping.test_slow` was added in a previous CL for #1367 . It computes the colour gradients in a random 500x297 pixel image using an intermediate table of about 1.5M rows with 9 element partitions, each sorted on two axes. It is also the test case that led me to work on this issue. Should I force parallelism and move it?

Reply:
Sure, please do.

Reply:
I've made a set of tests with about 1M rows that checks some boundary conditions around block sizes and copying strings across thread boundaries.  Coming shortly.

Reply:
Well that test appears to segfault under Linux 32-bit. Is it possible for someone to run it on that hardware? Win32 seems fine.

Reply:
The Linux 32-bit tests run on a 64-bit system, just with the FORCE_32_BIT flag enabled. You can check out the exact command [here](https://github.com/cwida/duckdb/blob/master/.github/workflows/main.yml#L353). When I ran into any failures on the CI there I always got the same failure running that script on a regular 64-bit Linux machine. Perhaps give that a shot (assuming you have access to a regular Linux box)?

Reply:
I don't but I'll see if I can make one of my Macs do it.  It's almost certainly something in the thread-safe copy code I missed.

Reply:
My Macs are all running Big Sur, which will not run 32 bit apps. Sigh.

Reply:
Perhaps try running with the thread sanitizer to see if that flags anything?

Reply:
Sadly no.  My Big Sur Mac won't link 32-bit app, but I have an old one at home I might be able to convince to reproduce it. A friend has an Ubuntu box I can borrow but not until next week.

Reply:
My old office iMac is too old to build with (10.10!) And the thread sanitiser didn't flag anything.

If anyone can have a quick look at this I'd appreciate it. I suspect that someone who has a Linux box and is familiar with the platform debuggers could find the problem pretty quickly. If not, I can probably muddle through some time next week.

Reply:
The issue shows up in debug builds and I think it is not caused by my CL.  if you look at the destructor for `WindowSegmentTree::~WindowSegmentTree`, it is doing pointer arithmetic assuming 64 bit pointers. When I look at the `addresses` array, the odd values are `0xffffffff`.

Reply:
That fixed it.